content/05.heterogeneity.md

```diff
@@ -1,3 +1,20 @@
+A key consideration here is, if possible, active dialogue with the data generators or experts in the field who may have unexpected insights into potential sources of variation. 
+One example of the value of this, experienced by the authors, occurred when studying tumors associated with the disease neurofibromatosis type 1. 
+These datasets were, unbeknownst to the computational biologists, generated from samples obtained with vastly different surgical techniques (laser ablation and excision vs standard excision), resulting in substantial biological differences that are a consequence of process, not reality. 
+Consequently, careful assessment of confounding factors and implementation of normalization methods is important to identifying biologically meaningful features within a dataset. 
+Assessment of confounding factors and heterogeneity in rare disease datasets is perhaps most easily performed using unsupervised learning approaches such as clustering and dimensionality reduction. 
+Clustering methods like k-means clustering or hierarchical clustering can be used to characterize the structure present in many different types of data such as genomic, imaging, and health records data to identify batch effects. 
+Similarly, a variety of dimensionality reduction methods are can be used to visualize sample heterogeneity and potential confounding variables, including multidimensional scaling (MDS), principal components analysis (PCA), t-distributed stochastic neighbor embedding (tSNE), and uniform manifold approximation and projection (UMAP), among many others. [@doi:10.1007/978-3-540-33037-0_14; @doi:10.1098/rsta.2015.0202; @https://lvdmaaten.github.io/publications/papers/JMLR_2008.pdf; @arXiv:1802.03426]
+All of these methods can be effectively used to identify batch effects and other structure in the data (cite), though some, like tSNE and UMAP, have parameters that can substantially affect the output, and thus the interpretation, of the analysis [@doi:10.23915/distill.00002;@arXiv:1802.03426]. 
+Therefore, successful application of these methods requires a sufficient understanding of the underlying method and parameter sweeping to get a clear picture of the structure of the underlying data. 
+Dimensionality reduction techniques are not restricted to 'omic' data - they can also be used in rare disease applications to characterize the structure and heterogenity of imaging data [@doi:10.1101/826974], mass cytometry data [@doi:10.1038/ncomms14825], and others.
+Once the nature of the non-biological heterogeneity has been established, different techniques can be used to correct the differences. 
+Common approaches to ameliorate non-biological effects include reprocessing the raw data using a single analysis pipeline if the data are obtained from different sources, application of batch correction methods [@doi:10.1093/biostatistics/kxj037; @doi:10.1093/nar/gku864], normalization of raw values (e.g. z-scores).
```

Comment:
suggest adding TMM normalization (https://doi.org/10.1186/gb-2010-11-3-r25)

Reply:
Also, wonder if we can recommend features for selection of specific batch correction methods or normalization methods over others.

Reply:
Hmm. I think method choice will be driven somewhat by downstream implementation. e.g. adding batch as a covariate in a linear model to determine differential expression is not useful if you want batch normalized counts.

Reply:
I don't have a good sense for whether  this is too specific of a conversation for this piece, though.

Reply:
In terms of features, I was thinking more along the lines of what makes one normalization method better than another (eg. z-score vs TMM) in rare disease use cases... it may be as simple as the PCA plot looks better with one versus the other, or one has been used previously in rare disease and the other hasn't. I was just curious if there is a good rule of thumb that we can propose... it is also highly possible that a good rule of thumb does not exist :)

Reply:
I'd invite the other co-authors to suggest something, because I'm not aware of a good rule of thumb for this! 

I think the true rule of thumb here is to use a consistent data model and to have a codebase of different methods that can easily handle that data model, so that you waste less time throwing all the various methods at the wall.